src/transferslot.cpp

```diff
@@ -433,6 +433,20 @@ void TransferSlot::doio(MegaClient* client)
                 continue;
             }
 
+            if (reqs[i]->status == REQ_FAILURE && reqs[i]->httpstatus == 200 && transfer->type == GET && transferbuf.isRaid())
+            {
+                // check if we got some data and the failure occured partway through the part chunk.  If so, best not to waste it, convert to success case with less data
+                HttpReqDL *downloadRequest = (HttpReqDL *)reqs[i];
```

Comment:
`static_cast`?

Reply:
And maybe just get a reference instead of a pointer?

Reply:
updated with static_cast.  I had followed the existing usage, but have swapped all of them now, total of 8 in the file.  I've left them as pointers, I think it's more natural to keep that the same on both sides.